src/components/AddEvent.tsx

```diff
@@ -88,90 +81,103 @@ const AddEvent = () => {
-        }
+  //confirmation handler for overlapping events
+  const confirmReplaceHandler = () => {
+    const lastDate = new Date(selectedDate).addDays(parseInt(recurrence));
+    const dates = createDatesArray(selectedDate, lastDate, whenToRecur);
+    let newState = [...data];
+    dates.forEach((date) => {
+      const existingDate = data.find((event) => event.date === date);
+      if (existingDate) {
+        existingDate.event.forEach((elem, i) => {
+          if (elem.start < newEvent.end && newEvent.start < elem.end) {
+            existingDate.event.splice(i, 1);
+          }
+        });
+        const index = (data as EventDate[]).indexOf(existingDate);
+        newState[index].event.push(newEvent);
+      } else {
+        newState.push({ date, event: [newEvent] });
       }
-      return dateArray;
-    };
+    });
+    setData(newState);
+    toast.success("You've successfully added a new event", {
+      position: toast.POSITION.TOP_CENTER,
+    });
+  };
+
+  //decline handler for overlapping events
+  const declineReplaceHandler = () => {
+    toggleModal();
+  };
 
+  const formSubmitHandler = async (e: React.FormEvent) => {
+    e.preventDefault();
+    let dates = [selectedDate];
+    //checking if task is going to repeat
     if (recurrence !== "no") {
-      const lastDate = new Date(date).addDays(parseInt(recurrence));
-      const dates = getDates(date, fromDateToString(lastDate));
-      setData((prevState) => {
-        let newState = [...prevState];
-        dates.forEach((date) => {
-          const existingDate = prevState.find((event) => event.date === date);
-          if (existingDate) {
-            const index = prevState.indexOf(existingDate);
-            newState[index].event.push(newEvent);
-          } else {
-            newState.push({ date, event: [newEvent] });
-          }
-        });
-        return newState;
-      });
-      toast.success("You've successfully added a new event", {
-        position: toast.POSITION.TOP_CENTER,
-      });
-    } else {
-      setData((prevState) => {
-        if (prevState.find((elem) => elem.date === date)) {
-          let newState = [...prevState];
-          const selectedDay = prevState.filter((elem) => elem.date === date);
-          const indexToSplice = prevState.indexOf(selectedDay[0]);
-          selectedDay[0].event.push(newEvent);
-          const newDay = { date, event: selectedDay[0].event };
-          newState.splice(indexToSplice, 1, newDay);
-          return newState;
-        }
-        return [...prevState, { date, event: [newEvent] }];
-      });
-      toast.success("You've successfully added a new event", {
-        position: toast.POSITION.TOP_CENTER,
-      });
+      const lastDate = new Date(selectedDate).addDays(parseInt(recurrence));
+      dates = createDatesArray(selectedDate, lastDate, whenToRecur);
     }
+
+    //checking for future tasks overlapping with current one
+    if (recurrenceOverlapCheck(dates, data, newEvent)) {
```

Comment:
bad name for the function. if it returns boolean, a good practice is to have _is_ in name. checkIsReccurenceOverlapped or simmilar